### DISCUSSION TITLE
Installation issue ( git not working, make missing a project file )

### DISCUSSION THREAD

**Discussion by achobanov** [mono/sdb#83] [Questions] [answered]
Hello,

I apologize if this is way stupid, but I am new to Linux, new to programming and C# in particular. On the flip side I don't like MonoDevelop at all and I wish to compile and debug manually. So many many thanks to all of you smart people making this possible, however I have an installation problem. I can not understand how to properly use the 'git' command and this is ( I presume ) causing make to yell that it is missing Project file "debugger-libs.sln". Would you mind to help out?

Thanks,
Alex

**Comment by DavidKarlas** [comment]
You probably need to run

```
git submodule update --init --recursive
```

Good luck!

**Comment by achobanov** [comment]
Yea, probably. The problem is that I can't. Outputs: fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

**Comment by alexrp** [comment, marked as answer]
Hey, are you building from a Git clone or a tarball? If the latter, perhaps GitHub doesn't include submodules in tarballs (which would suck).

At any rate, building from Git like this should work:

```
$ cd /home/<you>/src # or wherever
$ git clone git://github.com/mono/sdb.git
$ cd sdb
$ git submodule update --init --recursive
$ make
```

**Comment by achobanov** [comment]
That seemed to work, but I smell something fishy. In other words make check outputs:

fsharpc --debug+ --nologo --warnaserror --keyfile:mono.snk --out:chk/check.exe --target:exe chk/check.fs
make: fsharpc: Command not found
make: **\* [chk/check.exe] Error 127

If that helps. 

If you are familiar with the cs50 edX course, I did setup my system, including a make script that compiles C applications with clang. Could that interfere?

**Comment by alexrp** [comment]
That just means you don't have F# installed. Can't run the test suite without it.

**Comment by achobanov** [comment]
I tried installing F#, but it seems I was unsuccessful. I followed the steps in this tutorial:
http://www.cs.princeton.edu/courses/archive/fall10/cos597C/FsharpUbuntuInstall.htm

**Comment by alexrp** [comment]
I am not sure about what kind of system you are using, but on modern Debian/Ubuntu, you can just `apt-get install fsharp`. In any case, `sdb` should work fine without it.
